scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/model/validator/CatalogObjectValidator.java

```diff
@@ -175,6 +181,26 @@ private boolean matchKindAndContentType(CommonHttpResourceDownloader.ResponseCon
                                                             expectedContentType));
             }
         }
+        if (StringUtils.isNotEmpty(expectedBucketName)) {
+            String catalogObjBucketName = jsonNode.path("bucket_name").asText();
+            if (!catalogObjBucketName.contains(expectedBucketName) ||
+                (expectedBucketName.startsWith("%") && !catalogObjBucketName.startsWith(expectedBucketName)) ||
```

Comment:
if expectedBucketName starts with % the catalog object name can be any word ENDING with expectedBucketName minus the starting % character.

The contains will return false if catalogObjBucketName = pre_toto_post and expectedBucketName = %toto% 

Also, your use of OR expression means that you will throw a ValidationExpression at the first predicate that returns true.

Morality, you should first analyse the expected bucket name to see what kind of expression it is and then perform only 1 type of test.

Also consider introducing private methods as the test for bucket name and object name is similar